### ISSUE TITLE
WKB parsing error

### ISSUE THREAD

**Issue by dmh126** [Oslandia/py3dtiles#15]
Hello!

I'm trying to use this module to convert WKB -> B3dm and I always have an error when trying to parse a wkb file different than `test/building.wkb`.

At first I used code like that (it is an example from GDAL documentation), I just created a multipolygon feature and saved it as a wkb:

```
from osgeo import ogr

multipolygon = ogr.Geometry(ogr.wkbMultiPolygon)

# Create ring #1
ring1 = ogr.Geometry(ogr.wkbLinearRing)
ring1.AddPoint(0, 0)
ring1.AddPoint(5, 0)
ring1.AddPoint(5, 5)
ring1.AddPoint(0, 5)
ring1.AddPoint(0, 0)

# Create polygon #1
poly1 = ogr.Geometry(ogr.wkbPolygon)
poly1.AddGeometry(ring1)
multipolygon.AddGeometry(poly1)

wkb = multipolygon.ExportToWkb()
```
And the rest of this script I just copied from the **How to write a .b3dm file** section. It gives me an error:

```
Traceback (most recent call last):
  File "temp.py", line 77, in <module>
    ts = TriangleSoup.from_wkb_multipolygon(wkb)
  File "/.../py3dtiles/wkb_utils.py", line 27, in from_wkb_multipolygon
    multipolygons = [parse(bytes(wkb))]
  File "/.../py3dtiles/wkb_utils.py", line 170, in parse
    pntOffset]), dtype=np.float32)
struct.error: unpack requires a bytes object of length 16
```
This wkb file is valid, I can open it in FME viewer.
Then I tried to figure this out and I noticed, that `multipolygons = [parse(bytes(wkb))]` returns triangulated multipolygon (building.wkb):
```
[
    [
        [P1, P2, P3, P1],
        [Pi, Pj, Pk, Pi],
        ...
    ]
]
```
So I tried with a simple triangle. And the same error occured. 

Finally I created a new method in the TriangleSoup class `from_geojson_multipolygon`, where instead of parsing a binary file I provide just a geojson geometry (nested lists and points as numpy arrays). This works, b3dm file is generated but it seems to be invalid. 
I used a simple triangle: `[[0,0,0], [5,0,0], [5,5,0], [0,0,0]]`
And I can't display it in the Cesium. I also generated a tileset in FME with this single feature and it works. B3dm file looks a little bit different. 

Any ideas what can be wrong?

**Comment by Jeremy-Gaillard** [comment]
Hi,

After investigation, the problems seems to come from the endianness of the data. Currently, the parsing function assumes the byte order is little endian. Can you export your data in little endian until I push a fix?

**Comment by dmh126** [comment]
Thank you, exporting wkb with the different byte order works:

`wkb = multi_polygon.ExportToIsoWkb(byte_order=ogr.wkbNDR)`

One more question about this b3dm file. I can't visualize it in Cesium, I created a tileset and probably I'm doing something wrong.
I created wkb from a simple triangle: `[[0,0,0], [5,0,0], [5,5,0], [0,0,0]]` 
I set bbox to `[[0, 0, 0], [5, 5, 5]]`
Transformation matrix to `[[1, 0, 0, 0],
[0, 1, 0, 0],
[0, 0, 1, 0],
[0, 0, 0, 1]]`

And got a b3dm file. Now my question is, how to create a valid tileset.json, because it doesn't work with:
```
{
	"asset" : {
		"version" : "0.0"
	},
	"properties" : {
		"id" : {}
	},
	"geometricError" : 553790.0141403995,
	"root" : {
		"boundingVolume" : {
			"region" : [0, 0, 0.0872665, 0.0872665, 0, 1000 ]
		},
		"refine" : "add",
		"content" : {
			"boundingVolume" : {
				"region" : [ 0, 0, 0.0872665, 0.0872665, 0, 1000 ]
			},
			"url" : "data/mymodel.b3dm"
		}
	}
}
```
It should be displayed like that:
![triangle](https://user-images.githubusercontent.com/7457276/36254429-58e8da88-124b-11e8-876e-7ff753da3e26.png)

But there is no triangle on my Cesium globe.

**Comment by Jeremy-Gaillard** [comment]
You are missing a transform in your tileset. A common misconception is that defining a bounding region also implicitly defines the position of the underlying object, but it does not. The 3D Tiles specifications has a [section](https://github.com/AnalyticalGraphicsInc/3d-tiles#tile-transform) that explains quite well how to compute the transform.

**Comment by Jeremy-Gaillard** [comment]
This bug has been fixed by #16.

Note that your original example still won't work because 2D geometries are not yet supported for glTF conversion.
